packages/medusa/src/services/shipping-option.ts

```diff
@@ -0,0 +1,717 @@
+    try {
+      const option = await this.retrieve(optionId)
+
+      const optionRepo = this.manager_.getCustomRepository(
+        this.optionRepository_
+      )
+
+      return optionRepo.softRemove(option)
+    } catch (error) {
+      // Delete is idempotent, but we return a promise to allow then-chaining
+      return Promise.resolve()
+    }
+  }
+
+  /**
+   * Adds a requirement to a shipping option. Only 1 requirement of each type
+   * is allowed.
+   * @param {string} optionId - the option to add the requirement to.
+   * @param {ShippingOptionRequirement} requirement - the requirement for the option.
+   * @return {Promise} the result of update
+   */
+  async addRequirement(
+    optionId: string,
+    requirement: ShippingOptionRequirement
+  ): Promise<ShippingOption> {
+    return this.atomicPhase_(async (manager) => {
+      const option = await this.retrieve(optionId, {
+        relations: ["requirements"],
+      })
+      const validatedReq = await this.validateRequirement_(requirement)
+
+      if (option.requirements.find((r) => r.type === validatedReq.type)) {
+        throw new MedusaError(
+          MedusaError.Types.DUPLICATE_ERROR,
+          `A requirement with type: ${validatedReq.type} already exists`
+        )
+      }
+
+      option.requirements.push(validatedReq)
+
+      const optionRepo = manager.getCustomRepository(this.optionRepository_)
+      return optionRepo.save(option)
+    })
+  }
+
+  /**
+   * Removes a requirement from a shipping option
+   * @param {string} requirementId - the id of the requirement to remove
+   * @return {Promise} the result of update
+   */
+  async removeRequirement(
+    requirementId
+  ): Promise<ShippingOptionRequirement | void> {
+    return await this.atomicPhase_(async (manager) => {
+      const reqRepo: ShippingOptionRequirementRepository =
+        manager.getCustomRepository(this.requirementRepository_)
+
+      const requirement = await reqRepo.findOne({
+        where: { id: requirementId },
+      })
+      // Delete is idempotent, but we return a promise to allow then-chaining
+      if (typeof requirement === "undefined") {
+        return Promise.resolve()
+      }
+
+      return await reqRepo.softRemove(requirement)
+    })
+  }
+
+  /**
+   * Decorates a shipping option.
+   * @param {ShippingOption} optionId - the shipping option to decorate using optionId.
+   * @param {string[]} fields - the fields to include.
+   * @param {string[]} expandFields - fields to expand.
+   * @return {ShippingOption} the decorated ShippingOption.
+   */
+  async decorate(
+    optionId: string,
+    fields: (keyof ShippingOption)[] = [],
+    expandFields: (keyof ShippingOption)[] = []
+  ): Promise<ShippingOption> {
+    const requiredFields: (keyof ShippingOption)[] = ["id", "metadata"]
+
+    fields = fields.concat(requiredFields)
+
+    return await this.retrieve(optionId, {
+      select: fields,
+      relations: expandFields,
+    })
+  }
```

Comment:
todo: I believe this method is no longer used. If this is the case, we should remove it.